File: neatBracket.swift
import Foundation

if let input = readLine(){
    var balance = 0
    var isValid  = true

    for char in input{
        if char == "("{
            balance += 1
        }else if (char == ")"){
            balance -= 1
            if balance < 0{
                isValid = false
                break
            }
        }
    }
    if isValid && balance == 0{
        print("Yes")
    }else{
        print("No")
    }
}

File: neatBracket.cpp
#include<bits/stdc++.h>
using namespace std;
 
int main()
{
string s;
cin>>s;

int balance = 0;

for (char ch : s){
    if(ch == '('){
        balance++;
    }else if(ch == ')'){
        balance--;

        if(balance<0){
            cout<<"No"<<endl;
            return 0;
        }
    }

}

if(balance == 0){
    cout<<"Yes"<<endl;
}else{
    cout<<"No"<<endl;
}
    return 0;
}